src/telliot/submitter/base.py

```diff
@@ -69,26 +71,36 @@ def build_tx(self, value: bytes, request_id: str, gas_price: str) -> Any:
 
         return built_tx
 
-    def submit_data(self, value: bytes, request_id: str) -> Any:
+    def submit_data(
+        self, value: bytes, request_id: str, extra_gas_price: int = 0
+    ) -> Tuple[Any, SimpleNamespace]:
         """Submits data on-chain & provides a link to view the
         successful transaction."""
+        try:
+            status = SimpleNamespace(**{"error": None, "gas_price": None})
 
-        req = requests.get("https://ethgasstation.info/json/ethgasAPI.json")
-        prices = json.loads(req.content)
-        gas_price = str(prices["fast"])
-        print("retrieved gas price:", gas_price)
-        # gas_price = "3"
-        # print("gas price used:", gas_price)
+            req = requests.get("https://ethgasstation.info/json/ethgasAPI.json")
+            prices = json.loads(req.content)
+            gas_price = str(prices["fast"] + extra_gas_price)
+            status.gas_price = gas_price
 
-        tx = self.build_tx(value, request_id, gas_price)
+            tx = self.build_tx(value, request_id, gas_price)
 
-        tx_signed = self.acc.sign_transaction(tx)
+            tx_signed = self.acc.sign_transaction(tx)
 
-        tx_hash = self.endpoint.web3.eth.send_raw_transaction(tx_signed.rawTransaction)
+            tx_hash = self.endpoint.web3.eth.send_raw_transaction(
+                tx_signed.rawTransaction
+            )
 
-        tx_receipt = self.endpoint.web3.eth.wait_for_transaction_receipt(
-            tx_hash, timeout=360
-        )
-        print(f"View reported data: https://rinkeby.etherscan.io/tx/{tx_hash.hex()}")
+            tx_receipt = self.endpoint.web3.eth.wait_for_transaction_receipt(
+                tx_hash, timeout=360
+            )
+            print(
+                f"View reported data: https://rinkeby.etherscan.io/tx/{tx_hash.hex()}"
+            )
+
+            return tx_receipt, status

```

Comment:
I like @tallywiesenberg 's idea of standardizing responses (something like that in #94).  
Perhaps a combination of that and what you have here.